webview/src/experiments/components/table/body/Row.tsx

```diff
@@ -24,14 +25,15 @@ export const RowContent: React.FC<
     (state: ExperimentsState) => state.tableData.changes
   )
   const { getVisibleCells, original, index, getIsExpanded, subRows } = row
-  const { displayColor, error, starred, id, status, selected } = original
+  const { branch, displayColor, error, starred, id, status, selected } =
+    original
   const [firstCell, ...cells] = getVisibleCells()
   const isWorkspace = id === EXPERIMENT_WORKSPACE_ID
   const changesIfWorkspace = isWorkspace ? changes : undefined
 
   const { toggleRowSelected, selectedRows } = useContext(RowSelectionContext)
 
-  const isRowSelected = !!selectedRows[id]
+  const isRowSelected = !!selectedRows[getCompositeId(id, branch)]
```

Comment:
[Q] Should we call this `rowId` and put it on the experiment data?

Reply:
It was more complicated doing that as it needs to be added everywhere and it's only useful for row selection (something happening on the webview side only).